components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java

```diff
@@ -300,17 +300,22 @@ public int editBusinessRuleFromScratch(String uuid, BusinessRuleFromScratch busi
         int status = TemplateManagerConstants.OPERATION_FAILED;
 
         try {
-            derivedArtifacts = deriveArtifacts(businessRuleFromScratch);
+            derivedArtifacts = constructArtifacts(businessRuleFromScratch);
             deployableSiddhiApp = buildSiddhiAppFromScratch(derivedArtifacts, businessRuleFromScratch);
         } catch (TemplateManagerException e) {
-            log.error("Deriving artifacts for business rule while editing is failed due to " + e.getMessage());
+            log.error("Deriving artifacts for business rule: " + uuid + ", while editing is failed due to " + e.getMessage
+                    ());
```

Comment:
Please format this properly

Reply:
Fixed by 2c0119c4c48e1077512e685493252c388e68a520